src/engine/LoadingScreen.cs

```diff
@@ -104,6 +125,10 @@ public override void _Ready()
 
         spinner = GetNode<Control>("Spinner");
 
+        random = new Random();
```

Comment:
I think this should be initialized on line 38 to guard against "Target" being set before this is attached to the scene tree. I try to make my Control derived classes work fine if you first set all properties before attaching.